packages/csharp/PortingAssistant/PortingAssistant.Common/Utils/TelemetryCollectionUtils.cs

```diff
@@ -80,10 +82,85 @@ public static ProjectMetrics createProjectMetric(string runId, string triggerTyp
                 NumReferences = projectAnalysisResult.ProjectReferences.Count,
                 IsBuildFailed = projectAnalysisResult.IsBuildFailed,
                 CompatibilityResult = projectAnalysisResult.ProjectCompatibilityResult,
-                PortingAssistantVersion = MetricsBase.Version
+                PreCompatibilityResult = GenerateCompatibilityResults(preTriggerData?.sourceFileAnalysisResults?.ToList(),
```

Comment:
Why do we have to recalculate this? Isn't this already calculated in the client?

Reply:
I think client-ui do contain each apiResult.**codeEntityDetails**.  but the client-ui didn't sum up them by each CodeEntityType.  The calculation is done at PA.Client _porting-assistant-dotnet-client\src\PortingAssistant.Client.Analysis\AnalysisHandler.cs_ .  and later on it is passed to projectMetrics for uploading.  I didn't see the  projectAnalysisResult.ProjectCompatibilityResult returning back to client-ui.  Maybe I am wrong.